lib/message_request.js

```diff
@@ -1,176 +1,166 @@
+  return new Request(message).setOptions(options)
+}
 
-Request.buildUnsubscribe = function (scope) {
-  var message = { op: 'unsubscribe', to: scope};
-  return new Request(message);
-};
+Request.buildUnsubscribe = function (scope, message = { op: 'unsubscribe', to: scope }) {
+  return new Request(message)
+}
 
 // Instance methods
 
-Request.prototype.forceV2Sync = function (options) {
-  options = options || {};
-  options.version = 2;
-  this.setAttr('options', options);
-};
+Request.prototype.forceV2Sync = function (options = {}) {
+  options.version = 2
+  this.setAttr('options', options)
+}
 
 Request.prototype.setAuthData = function (configuration) {
-  this.setAttr('userData', configuration.userData);
+  this.setAttr('userData', configuration.userData)
   if (configuration.auth) {
-    this.setAttr('auth', configuration.auth);
-    this.setAttr('userId', configuration.userId);
-    this.setAttr('userType', configuration.userType);
-    this.setAttr('accountName', configuration.accountName);
+    this.setAttr('auth', configuration.auth)
+    this.setAttr('userId', configuration.userId)
+    this.setAttr('userType', configuration.userType)
+    this.setAttr('accountName', configuration.accountName)
   }
-};
+}
 
 Request.prototype.getMessage = function () {
-  return this.message;
-};
+  return this.message
+}
 
 Request.prototype.setOptions = function (options) {
   // Keep check for options, since it is sometimes purposefully null
   if (options) {
-    this.setAttr('options', options);
+    this.setAttr('options', options)
   }
 
-  return this;
-};
+  return this
+}
 
 Request.prototype.isPresence = function () {
-  return this.type === 'presence';
-};
+  return this.type === 'presence'
+}
 
 Request.prototype.setAttr = function (keyName, keyValue) {
-  this.message[keyName] = keyValue;
-};
+  this.message[keyName] = keyValue
+}
 
 Request.prototype.getAttr = function (keyName) {
-  return this.message[keyName];
-};
+  return this.message[keyName]
+}
 
 Request.prototype.payload = function () {
-  return JSON.stringify(this.getMessage());
-};
+  return JSON.stringify(this.getMessage())
+}
 
 Request.prototype.getType = function () {
-  return this.type;
-};
+  return this.type
+}
 
 // Private methods
 
 Request.prototype._isValid = function () {
   if (!this.message.op || !this.message.to) {
-    return false;
+    return false
   }
 
-  var type = this._getType();
+  const type = this._getType()
   if (type) {
     if (this._isValidType(type) && this._isValidOperation(type)) {
-      this.type = type;
-      return true;
+      this.type = type
+      return true
     }
   } else {
-    logger.error('missing type');
+    logger.error('missing type')
   }
-  return false;
-};
+  return false
+}
 
 Request.prototype._isValidType = function (type) {
-  for (var key in opTable) {
-    if (opTable.hasOwnProperty(key) && key === type) {
-      return true;
+  let key
+  for (key in opTable) {
```

Comment:
Could use directly in `for..in` syntax, since key is supposed to be a scoped variable:
```
for (let key in opTable) {
...
}
```

Reply:
👍 but standard complained it wasn't reassigned, to it's a `const` now